core/src/main/java/org/infinispan/commands/write/PrimaryAckCommand.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * @author Pedro Ruivo
+ * @since 9.0
+ */
+public class PrimaryAckCommand extends BaseRpcCommand {
+
+   public static final byte COMMAND_ID = 30;
+   private static final Type[] CACHED_TYPE = Type.values();
+   private CommandInvocationId commandInvocationId;
+   private Object returnValue;
+   private Type type;
+   private CommandAckCollector commandAckCollector;
+   private int topologyId;
+
+   public PrimaryAckCommand() {
+      super(null);
+   }
+
+   public PrimaryAckCommand(ByteString cacheName) {
+      super(cacheName);
+   }
+
+   public PrimaryAckCommand(ByteString cacheName, CommandInvocationId commandInvocationId, int topologyId) {
+      super(cacheName);
+      this.commandInvocationId = commandInvocationId;
+      this.topologyId = topologyId;
+   }
+
+   private static Type valueOf(int index) {
+      return CACHED_TYPE[index];
+   }
+
+   @Override
+   public Object invoke() throws Throwable {
+      ack();
+      return null;
+   }
+
+   @Override
+   public CompletableFuture<Object> invokeAsync() throws Throwable {
+      ack();
+      return CompletableFutures.completedNull();
+   }
+
+   @Override
+   public byte getCommandId() {
+      return COMMAND_ID;
+   }
+
+   @Override
+   public boolean isReturnValueExpected() {
+      return false;
+   }
+
+   @Override
+   public void writeTo(ObjectOutput output) throws IOException {
+      output.writeInt(topologyId);
+      CommandInvocationId.writeTo(output, commandInvocationId);
+      MarshallUtil.marshallEnum(type, output);
+      switch (type) {
+         case SUCCESS_WITH_RETURN_VALUE:
+         case UNSUCCESSFUL_WITH_RETURN_VALUE:
+            output.writeObject(returnValue);
+            break;
+         default:
+      }
+   }
+
+   @Override
+   public void readFrom(ObjectInput input) throws IOException, ClassNotFoundException {
+      topologyId = input.readInt();
+      commandInvocationId = CommandInvocationId.readFrom(input);
+      type = MarshallUtil.unmarshallEnum(input, PrimaryAckCommand::valueOf);
+      assert type != null;
+      switch (type) {
+         case SUCCESS_WITH_RETURN_VALUE:
+         case UNSUCCESSFUL_WITH_RETURN_VALUE:
+            returnValue = input.readObject();
+            break;
+         case SUCCESS_WITH_BOOL_RETURN_VALUE:
+            returnValue = true;
+            break;
+         case UNSUCCESSFUL_WITH_BOOL_RETURN_VALUE:
+            returnValue = false;
+            break;
+         default:
+      }
+   }
+
+   public void initWithReturnValue(boolean success, Object returnValue) {
+      this.returnValue = returnValue;
+      if (success) {
+         type = Type.SUCCESS_WITH_RETURN_VALUE;
+      } else {
+         type = Type.UNSUCCESSFUL_WITH_RETURN_VALUE;
+      }
+   }
+
+   public void initWithBoolReturnValue(boolean success) {
+      this.returnValue = success;
+      if (success) {
+         type = Type.SUCCESS_WITH_BOOL_RETURN_VALUE;
+      } else {
+         type = Type.UNSUCCESSFUL_WITH_BOOL_RETURN_VALUE;
```

Comment:
don't get what you mean. you don't like the enum and you'd like to  have a ```successful``` field instead. is that it?